File: summary/go/13_bfs.go
package main

import "fmt"

func breadthFirstSearch(graph map[string][]string, startVertex string) []string {
	queue := []string{startVertex}
	visited := make(map[string]bool)
	visited[startVertex] = true
	orderOfVisit := []string{}

	for len(queue) > 0 {
		currentVertex := queue[0]
		queue = queue[1:]
		orderOfVisit = append(orderOfVisit, currentVertex)

		for _, neighbor := range graph[currentVertex] {
			if !visited[neighbor] {
				queue = append(queue, neighbor)
				visited[neighbor] = true
			}
		}
	}

	return orderOfVisit
}

// Example usage:
func main() {
	graph := map[string][]string{
		"A": {"B", "C"},
		"B": {"A", "D", "E"},
		"C": {"A", "F"},
		"D": {"B"},
		"E": {"B", "F"},
		"F": {"C", "E"},
	}
	result := breadthFirstSearch(graph, "A")
	fmt.Println("Order of visit from A:", result)
}

File: summary/dart/13_bfs.dart
import 'dart:collection';

List<String> breadthFirstSearch(Map<String, List<String>> graph, String startVertex) {
  Queue<String> queue = Queue<String>();
  queue.add(startVertex);
  List<String> orderOfVisit = [];
  Map<String, bool> visited = {startVertex: true};

  while (queue.isNotEmpty) {
    String currentVertex = queue.removeFirst();
    orderOfVisit.add(currentVertex);

    for (String neighbor in graph[currentVertex]!) {
      if (!visited.containsKey(neighbor)) {
        queue.add(neighbor);
        visited[neighbor] = true;
      }
    }
  }

  return orderOfVisit;
}

void main() {
  Map<String, List<String>> graph = {
    'A': ['B', 'C'],
    'B': ['A', 'D', 'E'],
    'C': ['A', 'F'],
    'D': ['B'],
    'E': ['B', 'F'],
    'F': ['C', 'E']
  };

  List<String> result = breadthFirstSearch(graph, 'A');
  print('Order of visit: $result');
}
